### DISCUSSION TITLE
User selection for new TTGO ESP32-S3 with 170x320 display

### DISCUSSION THREAD

**Discussion by MartinNohr** [Bodmer/TFT_eSPI#2010] [Compatible displays and setup files]
Is there a user choice for the TTGO ESP32-S3 with 170x320 display. This is the new version of the TTGO T-Display.

**Comment by Bodmer** [comment]
I assume it is this board:

http://www.lilygo.cn/prod_view.aspx?TypeId=50062&Id=1411&FId=t3:50062:3

Which has an 8 bit paralel interface.

It should be possible to adapt this setup file by changing the pins used:

https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setups/Setup70d_ILI9488_S3_Parallel.h

**Reply by Bodmer** [reply]
I see the display uses GPIO pins that are higher than 31 so the library is not compatible at the moment.

**Reply by Bodmer** [reply]
It looks like a modified library is included here:

https://github.com/Xinyuan-LilyGO/T-Display-S3

**Comment by MartinNohr** [comment]
If that board becomes popular, and I think it will, then I would think these changes should be incorporated into the main code here.

**Reply by Bodmer** [reply]
See #1948

**Comment by gotnull** [comment]
Did we ever figure this out?

**Comment by Bodmer** [comment]
I created a temporary patch here:
https://github.com/Bodmer/TFT_eSPI/discussions/2134#discussioncomment-4081722

This allows high GPIO to be used. Let me know if it works for you.

The support will be added in the next library update, probably next week.

**Comment by Bodmer** [comment]
Support has been added to the branch here:

https://github.com/Bodmer/TFT_eSPI/tree/T-Display-S3

Please test and let me know if it works OK.  There is a setup file for the display:
https://github.com/Bodmer/TFT_eSPI/blob/T-Display-S3/User_Setups/Setup206_LilyGo_T_Display_S3.h

This has already been selected in the seup file so the library should run straight away with no edits:
https://github.com/Bodmer/TFT_eSPI/blob/T-Display-S3/User_Setup_Select.h#L131

**Comment by mcbp223** [comment]
I have this board too, have tested the branch in PlatformIO. With bouncing circles example it gives 26.5fps. With all defines copied in it, platformio.ini is like

```ini
[env:T-Display-S3]
# ESP32-S3R8 (8MB PSRAM)
# Winbond 25Q128JVSQ (16MB flash)
# 1.14" TFT 170x320, ST7789, 8-bit parallel

platform = espressif32
board = esp32-s3-devkitc-1
framework = arduino

board_build.boot = qio
board_build.psram_type = opi
board_build.arduino.memory_type = qio_opi

# 2 x 4.5MB app, 6.875MB SPIFFS
;board_build.partitions = large_spiffs_16MB.csv
;board_build.partitions = default_16MB.csv
board_build.partitions = default.csv

lib_deps = 
	SPI
	FS
	SPIFFS
	https://github.com/Bodmer/TFT_eSPI.git#T-Display-S3

build_flags =
	-D BOARD_HAS_PSRAM
	-D ARDUINO_USB_MODE=1
	-D ARDUINO_USB_CDC_ON_BOOT=1
	;
	-D USER_SETUP_LOADED=1
;	*** Setup206 _LilyGo_T_Display_S3.h ***
	-D ST7789_DRIVER
	-D CGRAM_OFFSET
	-D TFT_RGB_ORDER=TFT_BGR
	-D TFT_INVERSION_ON
	-D TFT_PARALLEL_8_BIT
	-D TFT_WIDTH=170
	-D TFT_HEIGHT=320
	-D TFT_DC=7
	-D TFT_RST=5
	-D TFT_WR=8
	-D TFT_RD=9
	;
	-D TFT_D0=39
	-D TFT_D1=40
	-D TFT_D2=41
	-D TFT_D3=42
	-D TFT_D4=45
	-D TFT_D5=46
	-D TFT_D6=47
	-D TFT_D7=48
	;
	-D TFT_BL=38
	-D TFT_BACKLIGHT_ON=HIGH
	;
	-D LOAD_GLCD
	-D LOAD_FONT2
	-D LOAD_FONT4
	-D LOAD_FONT6
	-D LOAD_FONT7
	-D LOAD_FONT8
	-D LOAD_GFXFF
	-D SMOOTH_FONT
```

Unrelated to display, when testing with 16MB partition schemes the board kept rebooting continuously.

**Reply by Bodmer** [reply]
Thanks for testing, I will merge soon.

**Comment by TangerineDreamer** [comment]
I have this one too,i'm glad you make an adaptation for the user setups,but it's not very perfect,
i use the TFT_clock sketch from the 160x128 repertory,here is the result as is (no change done to your setup) :
![T-Display S3 test](https://user-images.githubusercontent.com/76820158/212359731-1a086014-2fa2-4834-a2ec-5fc8d81829b0.jpg)

seems it keeps from the last memory sketch: graphic test one lib

**Reply by Bodmer** [reply]
Comment out this line and try again:
https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setups/Setup206_LilyGo_T_Display_S3.h#L6

**Comment by TangerineDreamer** [comment]
Seems it's good now (you know the colors of the sketch):
![T-Display S3 test good](https://user-images.githubusercontent.com/76820158/212369118-acf7969f-92ec-4240-9ac1-bb1e2a38bfda.jpg)

**Comment by TangerineDreamer** [comment]
here's the benshmark results:
![Benshmark](https://user-images.githubusercontent.com/76820158/212371567-1a2a55c4-7031-4f39-9b29-3d112d4426bf.jpg)

**Comment by Bodmer** [comment]
I have updated this file as there was a command missing:
https://github.com/Bodmer/TFT_eSPI/blob/master/TFT_Drivers/ST7789_Init.h

With this file replaced the line below can be reinstated and should improve the colour rendition:
https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setups/Setup206_LilyGo_T_Display_S3.h#L6

The test results look good (that test clears the screen 60 times, hence the large clear screen time.

**Comment by TangerineDreamer** [comment]
hay bodmer,it works if the line:
#define INIT_SEQUENCE_3 is commented,so for me the good result is:
//#define INIT_SEQUENCE_3

**Comment by TangerineDreamer** [comment]
It's ok now with your very last update:
#define INIT_SEQUENCE_3 is let as is: uncomment
benshmark is same as before,glad that works :)

**Comment by eliliam** [comment]
Is this close to being merged? This would let us move forward over in ESPHome to add support for the board as well as its display.

**Comment by Bodmer** [comment]
Yes, it is in the master.
